tools/src/main/java/com/navercorp/pinpoint/tools/NetworkAvailabilityChecker.java

```diff
@@ -57,24 +56,94 @@ public static void main(String[] args) {
             return;
         }
 
-        ThriftTransportConfig thriftTransportConfig = profilerConfig.getThriftTransportConfig();
-        try {
-            checkUDPStat(thriftTransportConfig);
-        } catch (Exception e) {
-            e.printStackTrace();
+        if (profilerConfig.getTransportModule().toString().equals("GRPC")) {
+
+            System.out.println("Transport Module set to GRPC");
+
+            GrpcTransportConfig grpcTransportConfig = new GrpcTransportConfig();
+            grpcTransportConfig.read(profilerConfig);
+
+            try {
+                checkGRPCBase(grpcTransportConfig);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            try {
+                checkGRPCMeta(grpcTransportConfig);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            try {
+                checkGRPCStaT(grpcTransportConfig);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            try {
+                checkGRPCSpan(grpcTransportConfig);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+
+        } else {
+
+            System.out.println("Transport Module set to THRIFT");
+
+            ThriftTransportConfig thriftTransportConfig = profilerConfig.getThriftTransportConfig();
+            try {
+                checkUDPStat(thriftTransportConfig);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            try {
+                checkUDPSpan(thriftTransportConfig);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            try {
+                checkTCP(thriftTransportConfig);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
         }
 
-        try {
-            checkUDPSpan(thriftTransportConfig);
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
 
-        try {
-            checkTCP(thriftTransportConfig);
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
+    }
+
+    private static void checkGRPCBase(GrpcTransportConfig grpcTransportConfig) throws Exception {
+        String ip = grpcTransportConfig.getAgentCollectorIp();
+        int port = grpcTransportConfig.getAgentCollectorPort();
+
+        NetworkChecker checker = new TCPChecker("TCP Base", ip, port);
+        checker.check();
+    }
+
+    private static void checkGRPCMeta(GrpcTransportConfig grpcTransportConfig) throws Exception {
+        String ip = grpcTransportConfig.getMetadataCollectorIp();
+        int port = grpcTransportConfig.getMetadataCollectorPort();
+
+        NetworkChecker checker = new TCPChecker("TCP Meta", ip, port);
+        checker.check();
+    }
+
+    private static void checkGRPCStaT(GrpcTransportConfig grpcTransportConfig) throws Exception {
```

Comment:
I think "checkGRPCStat" is better than "checkGRPCStaT" :)

Reply:
got it